File: js/task1/14mod1.js
function towers(str, x) {
    let n = str.length;
    let cover = new Array(n).fill(false); // Массив с false значениями
    for (let i = 0; i < n; i++) {
        if (str[i] === '1') {
            cover[i] = true;
        } else {
            cover[i] = false;
        }
    }
    for (let i = 0; i < n; i++) {
        if (cover[i] === true) {
            for (let k = 0; k <= x; k++) {
                if (i + k < n) cover[i + k] = true;  // Покрываем вправо
                if (i - k >= 0) cover[i - k] = true;  // Покрываем влево
            }
        }
    }
    for (let i = 0; i < n; i++) {
        if (!cover[i]) {
            return false;
        }
    }
    return true;
}

const readline = require('readline'); // Модуль для взаимодействия через консоль

const rl = readline.createInterface({ // Инициализируем взаимодействие с пользователем
    input: process.stdin,
    output: process.stdout
});

rl.question("Введите вышки: ", function(s) {
    rl.question("Введите показатель x: ", function(x) {
        x = parseInt(x); // Преобразуем x в число
        if (towers(s, x)) {
            console.log("Yes");
        } else {
            console.log("No");
        }
        rl.close(); // Закрываем интерфейс после выполнения
    });
});

File: ruby/task1/14mod1.rb
def towers(str, x)
    n = str.length
    cover = Array.new(n, false)
    (0...n).each do |i| # Цикл for
      cover[i] = true if str[i] == '1'
    end
    (0...n).each do |i|
      if str[i] == '1'
        (-x..x).each do |k|
          pos = i + k
          cover[pos] = true if pos >= 0 && pos < n
        end
      end
    end
    cover.all? # Условие (true/false если все 1)
  end
  
  # Основная программа
  puts "Введите вышки:"
  s = gets.chomp  # Читаем строку
  puts "Введите показатель x:"
  x = gets.chomp.to_i  # Преобразуем ввод в целое число
  # Вывод
  if towers(s, x)
    puts "Yes"
  else
    puts "No"
  end
  